docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -110,6 +110,36 @@ We need to be able to process transactions and roll-back state updates if a tran
 
 We identified use-cases, where modules will need to save an object commitment without storing an object itself. Sometimes clients are receiving complex objects, and they have no way to prove a correctness of that object without knowing the storage layout. For those use cases it would be easier to commit to the object without storing it directly.
 
+
+### Low Level Interface
+
+Modules should be able to commit a value fully managed by the module itself. For example, a module can manage its own special database and commit its state by setting a value only to `SC`.
+Similarly, a module can save a value without committing it - this is useful for ORM module or secondary indexes (eg x/staking `UnbondingDelegationKey` and `UnbondingDelegationByValIndexKey`).
+
+Currently, a module can access a store only through `sdk.Context`. We add the following methods to the `sdk.Context`:
+
+```
+type StoreAccess inteface {
+    KVStore(key []byte) KVStore  // reads and writes to combined store (SS & SC)
+    InternalSCStore(key []byte) KVStore  // reads and writes only to the SC
+    InternalSSStore(key []byte) KVStore  // reads and writes only to the SS
+}
```

Comment:
During ADR-040 call, we were thinking about separating SS and  InternalSS into separate namespaces. This way we avoid race conditions and conflicts we discuss below.

If we separate in different namespaces then we should rename _Internal SS_ to something else, eg: `PlainSSKVStore` or `StateKVStore`.

Reply:
When thinking more about it, I think it will make more sense than the low level interface we have been discussing.

Reply:
I think it's fine to present both option 1 & 2 for exposing SS/SC, but as I've said before it simply shouldn't be on `sdk.Context` but rather `MultiStore` and the parameter type should be `StoreKey` not `[]byte`. I can approve if this change is made but I still consider this very much `PROPOSED` status.